### DISCUSSION TITLE
How to submit the form outside the Component.

### DISCUSSION THREAD

**Discussion by mahadih-gg** [react-hook-form/react-hook-form#7131] [Q&A]
# Is there any way to triggered  the handleSubmit() function by state or any custom function?

**Comment by bluebill1049** [comment]
https://react-hook-form.com/api/useform/handlesubmit

```tsx
// It can be invoked remotely as well
handleSubmit(onSubmit)();

// You can pass an async function for asynchronous validation.
handleSubmit(async (data) => await fetchAPI(data))
```

**Reply by mahadih-gg** [reply]
![image](https://user-images.githubusercontent.com/68747533/143243843-54869094-edb4-45d0-9185-6ed0f5e62b67.png)
